openproblems/tasks/_cell_cell_communication/_common/metrics/odds_ratio.py

```diff
@@ -19,14 +18,21 @@ def odds_ratio(adata, merge_keys, top_n=100):
     a[0:top_n] = 1
     gt.loc[:, ["top_n"]] = a
 
-    # Shape to contingency table
-    table = np.array(gt.pivot_table(index=["top_n", "response"], aggfunc="size"))
+    top = gt[gt["top_n"] == 1]
+    tp = np.sum(top.response == 1)
+    fp = np.sum(top.response == 0)
 
-    # if positive or negative class is not in top_n
-    if table.shape != (4,):
-        return 1
+    bot = gt[gt["top_n"] == 0]
+    fn = np.sum(bot.response == 1)
+    tn = np.sum(bot.response == 0)
 
-    # Fisher ET
-    oddsratio, _ = stats.fisher_exact(table.reshape(2, 2))
```

Comment:
Unclear to me why the `oddsratio` code is changed in this PR, but happy to approve if you think this is cleaner

Reply:
Makes sense. IMO this probably would have been better as a separate PR but no reason to change it now

Reply:
Yeah you are right, but at this stage I guess better to rename the PR, as it became the general patch of what follow from the PR discussion